geode-common/src/main/java/org/apache/geode/internal/Retry.java

```diff
@@ -82,7 +84,9 @@ public void sleep(long sleepTime, TimeUnit sleepTimeUnit) throws InterruptedExce
       if (predicate.test(value)) {
         return value;
       } else {
-        timer.sleep(interval, intervalUnit);
+        long sleepTimeInNano =
+            Math.min(NANOSECONDS.convert(interval, intervalUnit), until - timer.nanoTime());
```

Comment:
I recommend converting this outside the loop since it never changes.

Reply:
no, it changes, only in the last iteration though.

Reply:
ah yes!